libass/ass.c

```diff
@@ -1535,17 +1535,33 @@ ASS_Track *ass_new_track(ASS_Library *library)
 
 int ass_track_set_feature(ASS_Track *track, ASS_Feature feature, int enable)
 {
-    switch (feature) {
-    case ASS_FEATURE_INCOMPATIBLE_EXTENSIONS:
-        //-fallthrough
+    if (feature >= sizeof(track->parser_priv->feature_flags) * CHAR_BIT || feature < 0)
+        return -1;
+
+    // all supported non-meta features
+    static const uint32_t supported = 0
 #ifdef USE_FRIBIDI_EX_API
-    case ASS_FEATURE_BIDI_BRACKETS:
-        track->parser_priv->bidi_brackets = !!enable;
+        | FEATURE_MASK(ASS_FEATURE_BIDI_BRACKETS)
 #endif
-        return 0;
+        ;
```

Comment:
This lone `;` is hurting my eyes. Maybe simple:
```c
    uint32_t supported = 0;
#ifdef USE_FRIBIDI_EX_API
    supported |= FEATURE_MASK(ASS_FEATURE_BIDI_BRACKETS);
#endif
```
would be better?

Reply:
That would be more repetitive than the current version, so I think the current approach is better for scaling with future additions. How about adding either a trailing `| 0` or enclosing the whole expression in cosmetic parantheses `()` to avoid a freestanding `;`? WIth the parantheses it would end in `);` similar to how structs end in `};`.

Reply:
I think that dynamic approach is better for scaling instead (it can even support runtime feature detection).
Otherwise I think that `0;` is better than `);`.

Reply:
Now using `0;`.

The availability of currently planned features are all determined at build-time, if we get runtime-dependent features in the futures we can still add them with `if (…) supported |= …`.
If we we're always using `supported |=` it would be a lot more repetitive though — or a bit more repetitive and less consistent imho; compare:
```C
// supported |= for build-time-determined features
    uint32_t supported = FEATURE_MASK(ASS_FEATURE_UNCOND1)
        | FEATURE_MASK(ASS_FEATURE_UNCOND2);
#ifdef USE_FRIBIDI_EX_API
    supported |= FEATURE_MASK(ASS_FEATURE_BIDI_BRACKETS);
#endif
    if (runtime_check()) {
        supported |= FEATURE_MASK(ASS_FEATURE_RUNTIME1);
    }


// Single expression for build-time-determined and unconditional features
    uint32_t supported =
#ifdef USE_FRIBIDI_EX_API
        FEATURE_MASK(ASS_FEATURE_BIDI_BRACKETS) |
#endif
        FEATURE_MASK(ASS_FEATURE_UNCOND1) |
        FEATURE_MASK(ASS_FEATURE_UNCOND2);
    if (runtime_check()) {
        supported |= FEATURE_MASK(ASS_FEATURE_RUNTIME1);
    }
```